modules/i3s/src/lib/parsers/parse-zip/end-of-central-directory.ts

```diff
@@ -0,0 +1,78 @@
+import {FileProvider} from './file-provider';
+import {ZipSignature, searchFromTheEnd} from './search-from-the-end';
+
+/**
+ * End of central directory info
+ * according to https://en.wikipedia.org/wiki/ZIP_(file_format)
+ */
+export type ZipEoCDRecord = {
+  /** Relative offset of local file header */
+  cdStartOffset: bigint;
+  /** Relative offset of local file header */
+  cdRecordsNumber: bigint;
+};
+
+const eoCDSignature: ZipSignature = [0x50, 0x4b, 0x05, 0x06];
+const zip64EoCDLocatorSignature = Buffer.from([0x50, 0x4b, 0x06, 0x07]);
+const zip64EoCDSignature = Buffer.from([0x50, 0x4b, 0x06, 0x06]);
+
+const offsets = {
+  CD_RECORDS_NUMBER_OFFSET: 8n,
+  CD_START_OFFSET_OFFSET: 16n,
+
+  ZIP64_EOCD_START_OFFSET_OFFSET: 8n,
+
+  ZIP64_CD_RECORDS_NUMBER_OFFSET: 24n,
+  ZIP64_CD_START_OFFSET_OFFSET: 48n
+};
+
+/**
+ * Parses end of central directory record of zip file
+ * @param fileProvider - buffer containing whole array
```

Comment:
Does it work only with array or with file handle as well?
```suggestion
 * @param fileProvider - FileProvider instance
```